ledger/acctonline.go

```diff
@@ -732,14 +754,17 @@ func (ao *onlineAccounts) TopOnlineAccounts(rnd basics.Round, voteRnd basics.Rou
 			var roundOffsetError *RoundOffsetError
 			if !errors.As(err, &roundOffsetError) {
 				ao.accountsMu.RUnlock()
-				return nil, err
+				return nil, basics.MicroAlgos{}, err
 			}
 			// the round number cannot be found in deltas, it is in history
 			inMemory = false
 			err = nil
 		}
 
 		modifiedAccounts := make(map[basics.Address]*ledgercore.OnlineAccount)
+		// Online accounts that will not be valid in voteRnd. Used to calculate their total stake,
+		// to be removed from the total online stake if required (lower the upper bound of total online stake in voteRnd).
+		invalidOnlineAccounts := make(map[basics.Address]*ledgercore.OnlineAccount)
```

Comment:
nit, you really only need to track addr -> MicroAlgos

Reply:
not really.
A bunch of stuff is needed. stateproofId , rewardsBased, etc...

Reply:
AFAICT you only ever use the contents of invalidOnlineAccounts here:
```
        for _, oa := range invalidOnlineAccounts {
            totalOnlineStake = ot.SubA(totalOnlineStake, oa.MicroAlgos)
```
```diff
--- a/ledger/acctonline.go
+++ b/ledger/acctonline.go
@@ -770,7 +770,7 @@ func (ao *onlineAccounts) TopOnlineAccounts(rnd basics.Round, voteRnd basics.Rou
        modifiedAccounts := make(map[basics.Address]*ledgercore.OnlineAccount)
        // Online accounts that will not be valid in voteRnd. Used to calculate their total stake,
        // to be removed from the total online stake if required (lower the upper bound of total online stake in voteRnd).
-       invalidOnlineAccounts := make(map[basics.Address]*ledgercore.OnlineAccount)
+       invalidOnlineAccounts := make(map[basics.Address]basics.MicroAlgos)
        if inMemory {
            // Determine how many accounts have been modified in-memory,
            // so that we obtain enough top accounts from disk (accountdb).
@@ -789,7 +789,7 @@ func (ao *onlineAccounts) TopOnlineAccounts(rnd basics.Round, voteRnd basics.Rou

                    if !(d.VoteFirstValid <= voteRnd && voteRnd <= d.VoteLastValid) {
                        modifiedAccounts[addr] = nil
-                       invalidOnlineAccounts[addr] = accountDataToOnline(addr, &d, genesisProto)
+                       invalidOnlineAccounts[addr] = accountDataToOnline(addr, &d, genesisProto).MicroAlgos
                        continue
                    }

@@ -837,7 +837,7 @@ func (ao *onlineAccounts) TopOnlineAccounts(rnd basics.Round, voteRnd basics.Rou
                if !(data.VoteFirstValid <= voteRnd && voteRnd <= data.VoteLastValid) {
                    // If already exists it originated from the deltas, meaning its data is more recent
                    if _, ok := invalidOnlineAccounts[addr]; !ok {
-                       invalidOnlineAccounts[addr] = data
+                       invalidOnlineAccounts[addr] = data.MicroAlgos
                    }
                    continue
                }
@@ -899,8 +899,8 @@ func (ao *onlineAccounts) TopOnlineAccounts(rnd basics.Round, voteRnd basics.Rou
            return nil, basics.MicroAlgos{}, err
        }
        ot := basics.OverflowTracker{}
-       for _, oa := range invalidOnlineAccounts {
-           totalOnlineStake = ot.SubA(totalOnlineStake, oa.MicroAlgos)
+       for _, ualgos := range invalidOnlineAccounts {
+           totalOnlineStake = ot.SubA(totalOnlineStake, ualgos)
            if ot.Overflowed {
                return nil, basics.MicroAlgos{}, fmt.Errorf("TopOnlineAccounts: overflow in stakeOfflineInVoteRound")
            }
```

Reply:
however it does not matter, this was just a nit observation, no action necessary